pkg/machinery/config/types/runtime/extensions/service_config.go

```diff
@@ -0,0 +1,158 @@
+var (
+	_ config.ExtensionServiceConfig = &ServiceConfigV1Alpha1{}
+	_ config.Document               = &ServiceConfigV1Alpha1{}
+	_ config.Validator              = &ServiceConfigV1Alpha1{}
+)
+
+// ServiceConfigV1Alpha1 is a extensionserviceconfig document.
+//
+//	examples:
+//	  - value: extensionServiceConfigV1Alpha1()
+//	alias: ExtensionServiceConfig
+//	schemaRoot: true
+//	schemaMeta: v1alpha1/ExtensionServiceConfig
+type ServiceConfigV1Alpha1 struct {
+	meta.Meta `yaml:",inline"`
+	//   description: |
+	//     Name of the extension service.
+	//   schemaRequired: true
+	ServiceName string `yaml:"name"`
+	//   description: |
+	//     The config files for the extension service.
+	ServiceConfigFiles []ConfigFile `yaml:"configFiles,omitempty"`
+	//   description: |
+	//     The environment for the extension service.
+	ServiceEnvironment []string `yaml:"environment,omitempty"`
+}
+
+// ConfigFile is a config file for extension services.
+type ConfigFile struct {
+	//   description: |
+	//     The content of the extension service config file.
+	ConfigFileContent string `yaml:"content"`
+	//   description: |
+	//     The mount path of the extension service config file.
+	ConfigFileMountPath string `yaml:"mountPath"`
+}
+
+// NewServicesConfigV1Alpha1 creates a new siderolink config document.
+func NewServicesConfigV1Alpha1() *ServiceConfigV1Alpha1 {
+	return &ServiceConfigV1Alpha1{
+		Meta: meta.Meta{
+			MetaKind:       ServiceConfigKind,
+			MetaAPIVersion: "v1alpha1",
+		},
+	}
+}
+
+// Clone implements config.Document interface.
+func (e *ServiceConfigV1Alpha1) Clone() config.Document {
+	return e.DeepCopy()
+}
+
+// Validate implements config.Validatator interface.
+func (e *ServiceConfigV1Alpha1) Validate(validation.RuntimeMode, ...validation.Option) ([]string, error) {
+	if e.ServiceName == "" {
+		return nil, fmt.Errorf("name is required")
+	}
+
+	if len(e.ServiceConfigFiles) == 0 && len(e.ServiceEnvironment) == 0 {
+		if len(e.ServiceConfigFiles) == 0 {
+			return nil, fmt.Errorf("no config files found for extension %q", e.ServiceName)
+		}
+
+		if len(e.ServiceEnvironment) == 0 {
+			return nil, fmt.Errorf("no environment defined for extension %q", e.ServiceName)
+		}
+	}
+
+	for _, file := range e.ServiceConfigFiles {
+		if file.ConfigFileContent == "" {
+			return nil, fmt.Errorf("extension content is required for extension %q", e.ServiceName)
+		}
+
+		if file.ConfigFileMountPath == "" {
+			return nil, fmt.Errorf("extension mount path is required for extension %q", e.ServiceName)
+		}
+	}
+
+	return nil, nil
+}
+
+// Name implements config.ExtensionServiceConfig interface.
+func (e *ServiceConfigV1Alpha1) Name() string {
+	return e.ServiceName
+}
+
+// ConfigFiles implements config.ExtensionServiceConfig interface.
+func (e *ServiceConfigV1Alpha1) ConfigFiles() []config.ExtensionServiceConfigFile {
+	return xslices.Map(e.ServiceConfigFiles, func(c ConfigFile) config.ExtensionServiceConfigFile {
+		return &ConfigFile{
```

Comment:
that's a nit, but looks like we could either do `return &c` here, or we could define methods `Content()` and `MountPath()` on non-pointer `ConfigFile` so we can avoid this copying

we still need `.Map()` to convert to the interface type though